src/ports/owner/component.ts

```diff
@@ -24,34 +29,46 @@ export function createOwnersComponent(options: {
       )
     }
 
-    const parsedFilters: FetchOptions<OwnersFilters, OwnersSortBy> = {
-      ...filters,
-      sortBy: filters.sortBy as OwnersSortBy,
-    }
+    try {
+      const client = await database.getPool().connect()
+      const schemaName = await getDbSchema(client, {
+        contractAddress: filters.contractAddress,
+      })
 
-    const data: { nfts: OwnerFragment[] } = await subgraph.query(
-      getOwnersQuery(parsedFilters, false)
-    )
+      if (!schemaName) {
+        throw new HttpError('Contract not found', 404)
+      }
 
-    let count = 0
+      const parsedFilters: FetchOptions<OwnersFilters, OwnersSortBy> = {
+        ...filters,
+        sortBy: filters.sortBy as OwnersSortBy,
+      }
 
-    while (true) {
-      const countData: { nfts: { id: string }[] } = await subgraph.query(
-        getOwnersQuery({ ...parsedFilters, skip: count }, true)
+      const ownersQuery = getOwnersSQLQuery(schemaName, parsedFilters)
+      const ownersCountQuery = getOwnersSQLQuery(
+        schemaName,
+        parsedFilters,
+        true
       )
-      count += countData.nfts.length
-      if (countData.nfts.length < MAX_RESULTS) break
-    }
 
-    const results = data.nfts.map((owner: OwnerFragment) => ({
-      issuedId: owner.issuedId,
-      ownerId: owner.owner.id,
-      orderStatus: owner.searchOrderStatus,
-      orderExpiresAt: owner.searchOrderExpiresAt,
-      tokenId: owner.tokenId,
-    }))
+      const [owners, ownersCount] = await Promise.all([
+        client.query<OwnerDBRow>(ownersQuery),
+        client.query<OwnerCountDBRow>(ownersCountQuery),
+      ])
 
-    return { data: results, total: count }
+      const results = owners.rows.map((owner: OwnerDBRow) => ({
+        issuedId: owner.issued_id,
+        ownerId: owner.owner,
+        tokenId: owner.token_id,
+      }))
```

Comment:
WDYT about moving this logic into an adapter?

Reply:
My bad, I thought the format of the adapters was the same as the `marketplace-favorites-server`